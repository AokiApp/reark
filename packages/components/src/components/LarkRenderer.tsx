import React, { useContext } from "react";
import { LarkApiContext } from "../contexts/LarkApiContext";
import { BlockStoreProvider } from "../contexts/BlockStoreContext";
import { BlockComponent } from "./BlockComponent";

export const LarkRenderer: React.FC = () => {
  const { blocks } = useContext(LarkApiContext);

  if (!blocks || blocks.length === 0) return null;

  const rootBlock =
    blocks.find((block) => block.block_type === 1 && !block.parent_id) ||
    blocks[0];

  if (!rootBlock) return null;

  return (
    <BlockStoreProvider items={blocks}>
      <BlockComponent blockId={rootBlock.block_id} />
    </BlockStoreProvider>
  );
};
