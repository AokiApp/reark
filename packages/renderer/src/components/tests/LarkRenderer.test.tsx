import { render } from "@testing-library/react";
import { LarkRenderer } from "../LarkRenderer";

describe("LarkRenderer", () => {
  it("正常系スナップショットテスト", () => {
    const exampleBlock = {
      block_id: "ZIjadstYfoQVMjxXAwRjM0rVpVg",
      block_type: 1,
      children: ["K2jmdGqW1o0t8fx3CDNjV6vKpsf"],
    };
    const childBlock = {
      block_id: "K2jmdGqW1o0t8fx3CDNjV6vKpsf",
      block_type: 2,
      children: [],
      text: "テスト用子ブロック",
    };
    const { container } = render(
      <LarkRenderer initialData={{ blocks: [exampleBlock, childBlock] }} />,
    );
    expect(container).toMatchSnapshot();
  });
});
