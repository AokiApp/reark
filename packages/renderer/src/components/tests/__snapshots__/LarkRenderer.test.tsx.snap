// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LarkRenderer 正常系スナップショットテスト 1`] = `
<div>
  <div
    class="reark-block"
  >
    <div
      class="reark-page"
    >
      <div
        class="reark-page-title"
      >
        <div
          class="reark-text-block"
          style="text-align: left;"
        />
      </div>
      <div
        class="reark-block"
      >
        <div
          class="reark-text-block"
          style="text-align: left;"
        />
      </div>
    </div>
  </div>
</div>
`;
